provider/testdata/src/lib.rs

```diff
@@ -102,42 +84,48 @@
 
 extern crate alloc;
 
-// If you want this to be made public stable, file an issue on the ICU4X repository.
-#[cfg(feature = "metadata")]
-#[doc(hidden)]
-pub mod metadata;
+#[path = "../metadata.rs.data"]
+mod metadata;
```

Comment:
I would have a preference that you keep these as toml files for one PR and put the change to .data.rs as a second PR because

1. Better isolation of the data file format change
2. Allows further discussion on that subtopic